src/core/Localization/Localization.ts

```diff
@@ -165,6 +198,37 @@ export default class Localization {
      * @returns Translated string
      */
     get(key: TranslationKey, options?: TranslationOptions): string {
+        // Check if there's a mapped old key in swapConfig and if user provided custom translation with old key
+        if (this.#customTranslations) {
+            const oldKeyMapping = this.#keySwapConfig[key];
```

Comment:
You may also want to consider renaming `oldKeyMapping` to `oldKey`, that is what we get from the swap config.
```suggestion
            const oldKey = this.#keySwapConfig[key];
```